msal/src/main/java/com/microsoft/identity/client/internal/CommandParametersAdapter.java

```diff
@@ -413,16 +374,16 @@ public static SignUpStartCommandParameters createSignUpStartCommandParameters(
      * @return Command parameter object
      * @throws ClientException
      */
-    public static SignUpStartUsingPasswordCommandParameters createSignUpStartUsingPasswordCommandParameters(
+    public static SignUpStartCommandParameters createSignUpStartCommandParameters(
             @NonNull final NativeAuthPublicClientApplicationConfiguration configuration,
             @NonNull final OAuth2TokenCache tokenCache,
             @NonNull final String username,
-            @NonNull final char[] password,
+            @Nullable final char[] password,
```

Comment:
update comment (line 373) to reflect the fact that password is optional

Reply:
The password is nullable but is still required as Java does not support default values. Developers have to specify all parameters.

Reply:
This class is only used internally, but Kotlin classes (our state machine). In those cases, this field is optional. We've done the same for e.g. `scopes` parameter in other methods.

Reply:
Java does not have optional parameters. Optional parameters are simulated in Java by overloading functions. As this parameter is in the middle, it wouldn't be good style to overload function with reduced set of parameters.

Reply:
My point is that this method isn't called from any Java class, and so we don't have to consider the non-optionality of this parameter. You can leave it like this however, that's also fine.